R/smtp_send.R

```diff
@@ -169,38 +222,39 @@ smtp_send <- function(email,
       `-inline` = no_options()
     )
 
-  # Create the vector of arguments related
-  # to file attachments
+  # Create the vector of arguments related to file attachments
   attachment_args_vec <- create_attachment_args_vec(email = email)
 
-  # Clean up arguments and options; create the
-  # vector that's needed for `processx::run()`
+  # Clean up arguments and options; create the vector that's
+  # needed for `processx::run()`
   run_args <-
     run_args %>%
     prune_args() %>%
     create_args_opts_vec() %>%
-    append_attachment_args_vec(
-      attachment_args_vec = attachment_args_vec
-    )
+    append_attachment_args_vec(attachment_args_vec = attachment_args_vec)
+
+  if (dry_run) {
 
-  if (!debug) {
+    if ("-pass" %in% run_args) {
+      run_args[which(run_args == "-pass")[1] + 1] <- "*****"
```

Comment:
Remove `which`

Reply:
This is a common mistake for me but, in this case, we need the index position for the `"-pass"` value to get the position of the following item (by adding `1`).